src/names/common.h

```diff
@@ -477,4 +477,42 @@ class CNameCache
 
 };
 
+/* ************************************************************************** */
+/* CNamePendingData.  */
+
+/**
+ * Keeps track of name_new data for a pending name_firstupdate. This is
+ * serialized to the wallet so that the firstupdate can be broadcast
+ * between client runs.
+ */
+class CNamePendingData
+{
+private:
+    std::vector<unsigned char> vchToAddress;
+    std::vector<unsigned char> vchHex;
+    std::vector<unsigned char> vchRand;
+    std::vector<unsigned char> vchData;
+
+public:
+    ADD_SERIALIZE_METHODS;
+
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action) {
+        READWRITE(vchToAddress);
+        READWRITE(vchHex);
+        READWRITE(vchRand);
+        READWRITE(vchData);
+    }
+
+    inline const std::string getToAddress() { return ValtypeToString(vchToAddress); }
+    inline const std::string getHex() { return ValtypeToString(vchHex); }
+    inline const std::string getRand() { return ValtypeToString(vchRand); }
+    inline const std::string getData() { return ValtypeToString(vchData); }
```

Comment:
If I understand the code correctly, you are actually storing the address in its string representation in the `vchType`.  Similarly for hex and rand, which are stored as hex strings, right?

I think since you have this wrapper class and already need to convert between `string` and `vchType`, you should actually encode the data properly:  The address can be a `CScript`, and hex/rand can be the actual binary data in `vchType`, not the hex string.

Reply:
I'm down to this last one before I start doing more testing, but I'm a little confused. Do you want me to not return `std::string` and instead return standard internal types like `CScript` etc? Or are you saying I can leave it accepting/returning `std::string` but encode internally and serialize to the reference types (`CScript`, etc)?

Reply:
I've got it working the first way, accepting/returning `std::string` but serializing toAddress, txid, rand to `CScript`, `uint256`, `uint160` respectively. If you had something different in mind, let me know. For the way the UI code is using the wallet, it's easier for me to just use strings and it allows me to not have to mess with internal types in the Qt code. But I can change if required.

Reply:
Thanks!  Yes, that's exactly what I meant - storing and serialising as raw data.  Interfacing to the UI as strings is certainly fine.